### PR TITLE
Add an attribute listener fired even if the value doesn't change

### PR DESCRIPTION
**Pull request by mfucci**
This attribute is meant to be used by local APIs

### CODE REVIEW

File: src/matter/cluster/server/AttributeServer.ts [hunk 1]
```diff
@@ -21,11 +22,16 @@ export class AttributeServer<T> {
     }
 
     set(value: T) {
-        if (value === this.value) return;
+        if (value === this.value) {
+            this.listeners.forEach(listener => listener(value, value));
```
Comment (Apollon77) [inline src/matter/cluster/server/AttributeServer.ts:26]: sneaky ... was thinking ... stop ... should one not be this.value .... but yes both are the same *gg*

File: src/matter/interaction/SubscriptionHandler.ts [hunk 2]
```diff
@@ -28,7 +28,7 @@ export class SubscriptionHandler {
         private readonly minIntervalFloorMs: number,
         private readonly maxIntervalCeilingMs: number,
     ) {
-        attributes.forEach(({ attribute }) => attribute.addListener(this.listener));
+        attributes.forEach(({ attribute }) => attribute.addMatterListener(this.listener));
```
Comment (Apollon77) [inline src/matter/interaction/SubscriptionHandler.ts:31]: Should you bind the "this.listener" to "this"?

**Review by Apollon77 — COMMENTED** [review]
Why InteractionProtocol.ts is in here as "full new file"?

**Comment by mfucci** [comment]
InteractionProtocol.ts : probably a side-effect of the squash. This file is rename in the base branch (to InteractionProtocol to make it consistent) and it was modified when changing the copyright notice.
Removed.

File: src/matter/interaction/SubscriptionHandler.ts [hunk 3]
```diff
@@ -28,30 +29,31 @@ export class SubscriptionHandler {
...
     }
 
     async sendUpdate() {
         const now = Date.now();
         const timeSinceLastUpdateMs = now - this.lastUpdateTimeMs;
         if (timeSinceLastUpdateMs < this.minIntervalFloorMs) {
-            clearTimeout(this.intervalId);
-            this.intervalId = setTimeout(() => this.sendUpdate(), this.minIntervalFloorMs - timeSinceLastUpdateMs);
+            this.updateTimer.stop();
+            this.updateTimer = Time.get().getTimer(setTimeout(() => this.sendUpdate(), this.minIntervalFloorMs - timeSinceLastUpdateMs);
```
Comment (Apollon77) [inline src/matter/interaction/SubscriptionHandler.ts:42]: should this not also be changed toTime.getTimer(...).start()... and also without a setTimeout in it?

File: src/matter/interaction/SubscriptionHandler.ts [hunk 4]
```diff
@@ -28,30 +29,31 @@ export class SubscriptionHandler {
...
         }
 
         const values = this.attributes.map(({ attribute, path }) => ({ path, valueVersion: attribute.getWithVersion(), template: attribute.template }));
         await this.sendUpdateMessage(values);
         this.lastUpdateTimeMs = now;
 
-        clearTimeout(this.intervalId);
-        this.intervalId = setTimeout(() => this.sendUpdate(), this.maxIntervalCeilingMs);
+        clearTimeout(this.updateTimer);
+        this.updateTimer = setTimeout(() => this.sendUpdate(), this.maxIntervalCeilingMs);
```
Comment (Apollon77) [inline src/matter/interaction/SubscriptionHandler.ts:51]: also here use timer.stop() to stop and not settimeout? (or is that not based on new Time stuff?)

File: src/matter/interaction/SubscriptionHandler.ts [hunk 5]
```diff
@@ -28,30 +29,31 @@ export class SubscriptionHandler {
...
     }
 
     cancel() {
-        this.attributes.forEach(({ attribute }) => attribute.removeListener(this.listener));
-        clearTimeout(this.intervalId);
+        this.attributes.forEach(({ attribute }) => attribute.removeMatterListener(this.listener));
+        clearTimeout(this.updateTimer);
```
Comment (Apollon77) [inline src/matter/interaction/SubscriptionHandler.ts:56]: same here this.updateTimer.stop()

**Review by Apollon77 — CHANGES_REQUESTED** [review]

**Comment by mfucci** [comment]
I merged the changes from the base branch and the random unrelated changes are not there anymore. Sorry about that.
Now the PR should be clean

**Review by Apollon77 — APPROVED** [review]
LGTM

**Pull request merged**